src/nupic/algorithms/Anomaly.cpp

```diff
@@ -0,0 +1,159 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+
+#include <vector>
+#include <numeric>
+#include <algorithm>
+#include <set>
+
+#include "nupic/algorithms/Anomaly.hpp"
+#include "nupic/utils/Log.hpp"
+//#define NTA_ASSERT(condition) if (condition) {} else throw "Error";
+
+using namespace std;
+
+namespace nupic
+{
+  namespace algorithms
+  {
+    namespace anomaly
+    { //FIXME for some reason googletests fail if using namespace nupic::algorithms::anomaly; is used instead!
```

Comment:
Can you explain this more? Do you expect to be able to replace the three namespaces with this?

```
using namespace nupic::algorithms::anomaly;
```

I would not expect the above to work since that just brings tokens into scope. It doesn't effect the namespace scope of things you define.